### PR TITLE
docs: improve LSP6 Natspec comments + add auto-generated docs

### PR DESCRIPTION
**Pull request by CJ42**
# What does this PR introduce?

## 📄 Documentation

Improve Natspec comments for `LSP6KeyManager` and its own modules + interfaces and add auto-generated docs.

### CODE REVIEW

File: dodoc/config.ts
Comment (b00ste) [inline dodoc/config.ts:1]: Let's update the stuff related to dodoc in a different PR. #630

File: package-lock.json
Comment (b00ste) [inline package-lock.json:1]: Came comment here. PR #630

File: package.json
Comment (b00ste) [inline package.json:1]: Came comment here. PR #630

File: contracts/LSP6KeyManager/ILSP6KeyManager.sol [hunk 1]
```diff
@@ -12,10 +12,10 @@ interface ILSP6KeyManager is
     /* is ERC165 */
 {
     /**
-     * @dev Emitted when a calldata payload that includes `selector` and `value` as msg.value was verified for `signer`
-     * @param signer the address of the controller that executed the calldata payload.
+     * @dev Emitted when the LSP6KeyManager contract verified the permissions of the `signer` successfully.
+     * @param signer the address of the controller that executed the calldata payload (either directly via {execute} or via meta transaction using {executeRelayCall}).
      * @param value the amount of native token to be transferred in the calldata payload.
-     * @param selector the bytes4 function of the function to run in the calldata payload.
+     * @param selector the bytes4 function of the function that was executed on the linked {target}
```
Comment (magalimorin18) [inline contracts/LSP6KeyManager/ILSP6KeyManager.sol:18]:
```suggestion
     * @param selector : the bytes4 function of the function that was executed on the linked {target}
```

File: contracts/LSP6KeyManager/ILSP6KeyManager.sol [hunk 2]
```diff
@@ -24,52 +24,64 @@ interface ILSP6KeyManager is
...
         address from,
         uint128 channelId
     ) external view returns (uint256);
 
     /**
-     * @notice execute the following payload on the ERC725Account: `payload`
-     * @dev the ERC725Account will return some data on successful call, or revert on failure
-     * @param payload the payload to execute. Obtained in web3 via encodeABI()
-     * @return the data being returned by the ERC725 Account
+     * @notice execute the following payload on the linked contract: `payload`
```
[the remark this answers is not in the document]
Comment (magalimorin18) [reply contracts/LSP6KeyManager/ILSP6KeyManager.sol:48]: FMI: what is the linked contract ? :)

File: contracts/LSP6KeyManager/ILSP6KeyManager.sol (hunk 1 above)
Comment (magalimorin18) [reply contracts/LSP6KeyManager/ILSP6KeyManager.sol:18]: Maybe nice to add a semi colon between the variable and the description ? What do you think ?

File: contracts/LSP6KeyManager/LSP6Errors.sol [hunk 3]
```diff
@@ -2,131 +2,157 @@
...
+ *
+ * @param value The value that is not a valid `bytes32[CompactBytesArray]`
+ * @param context A brief description of where the error occured.
  */
 error InvalidEncodedAllowedERC725YDataKeys(bytes value, string context);
 
 /**
- * @dev a `from` address is not allowed to have 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffff
- * in its list of AddressPermissions:AllowedCalls:<address>, as this allows any STANDARD:ADDRESS:FUNCTION.
+ * @dev Reverts when verifying the permissions of a `from` address for its allowed calls, and has a "any whitelisted call" allowed call set.
```
Comment (magalimorin18) [inline contracts/LSP6KeyManager/LSP6Errors.sol:119]:
```suggestion
 * @dev Reverts when a `from` address has "any whitelisted call" allowed call set. This revert happens during the verification of the permissions of the address for its allowed calls.
```

File: contracts/LSP6KeyManager/LSP6Errors.sol [hunk 4]
```diff
@@ -2,131 +2,157 @@
...
+ * A `from` address is not allowed to have 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffff
+ * in its list of `AddressPermissions:AllowedCalls:<address>`, as this allows any STANDARD:ADDRESS:FUNCTION.
  * This is equivalent to granting the SUPER permission and should never be valid.
  *
- * @param from the address that has any allowed calls whitelisted.
+ * @param from The controller address that has any allowed calls whitelisted set.
```
Comment (magalimorin18) [inline contracts/LSP6KeyManager/LSP6Errors.sol:124]:
```suggestion
 * @param from The controller address that has "any allowed calls" whitelisted set.
```

File: contracts/LSP6KeyManager/LSP6Errors.sol [hunk 5]
```diff
@@ -137,31 +163,34 @@ error LSP6BatchInsufficientValueSent(uint256 totalValues, uint256 msgValue);
...
- * @dev ERC725X operation type 4 (DELEGATECALL) is disallowed by default
+ * @dev Reverts when trying to do a `delegatecall` via the ERC725X.execute(uint256,address,uint256,bytes) (operation type 4)
+ * function of the linked {target}.
+ * `DELEGATECALL` is disallowed by default on the LSP6KeyManager.
  */
 error DelegateCallDisallowedViaKeyManager();
 
 /**
- * @dev reverts when the payload is invalid.
+ * @dev Reverst when the payload is invalid.
```
Comment (magalimorin18) [inline contracts/LSP6KeyManager/LSP6Errors.sol:173]:
```suggestion
 * @dev Reverts when the payload is invalid.
```

File: contracts/LSP6KeyManager/LSP6Errors.sol [hunk 6]
```diff
@@ -137,31 +163,34 @@ error LSP6BatchInsufficientValueSent(uint256 totalValues, uint256 msgValue);
...
  */
 error InvalidPayload(bytes payload);
 
 /**
- * @dev reverts when sending value to the `setData(..)` functions
+ * @dev Reverts when trying to call to the `setData(byte32,bytes)` or `setData(bytes32[],bytes[]) functions
```
Comment (magalimorin18) [inline contracts/LSP6KeyManager/LSP6Errors.sol:178]:
```suggestion
 * @dev Reverts when calling the `setData(byte32,bytes)` or `setData(bytes32[],bytes[]) functions
```

File: contracts/LSP6KeyManager/LSP6KeyManagerInit.sol [hunk 7]
```diff
@@ -18,8 +18,11 @@ contract LSP6KeyManagerInit is LSP6KeyManagerInitAbstract {
     }
 
     /**
-     * @notice Initiate the account with the address of the ERC725Account contract and sets LSP6KeyManager InterfaceId
-     * @param target_ The address of the ER725Account to control
+     * @notice Deploying a LSP6KeyManager linked to contract at address `target_`.
```
Comment (magalimorin18) [inline contracts/LSP6KeyManager/LSP6KeyManagerInit.sol:21]:
```suggestion
     * @notice Deploying a LSP6KeyManager linked to the contract with address `target_`.
```

**Comment by magalimorin18** [comment]
Very cool 
I left comments but it s mainly suggestions, you can decide to take them into account or not Thanks Jean 🎉

File: contracts/LSP6KeyManager/ILSP6KeyManager.sol [hunk 8]
```diff
@@ -24,52 +24,64 @@ interface ILSP6KeyManager is
...
     ) external payable returns (bytes memory);
 
     /**
-     * @dev batch `execute(bytes)`
+     * @dev Same as {execute} but execute a batch of payloads (abi-encoded function calls) in a single transaction.
+     *
+     * @param values An array of amount of native tokens to be transferred for each `payload`.
+     * @param payloads An array of abi-encoded function calls to execute successively on the linked {target}.
+     *
+     * @return An array of abi-decoded of return data returned by the functions called on the linked {target}.
```
[the remark this answers is not in the document]
Comment (CJ42) [reply contracts/LSP6KeyManager/ILSP6KeyManager.sol:67]: Nice catch! 👌🏻

File: contracts/LSP6KeyManager/ILSP6KeyManager.sol [hunk 9]
```diff
@@ -24,52 +24,64 @@ interface ILSP6KeyManager is
...
 
     /**
-     * @dev allows anybody to execute given they have a signed message from an executor
-     * @param signature bytes32 ethereum signature
-     * @param nonce the address' nonce (in a specific `_channel`), obtained via `getNonce(...)`. Used to prevent replay attack
-     * @param validityTimestamps two `uint128` timestamps concatenated, the first timestamp determines from when the payload can be executed, the second timestamp delimits the end of the validity of the payload. If `validityTimestamps` is 0, the checks regardin the timestamps are skipped
-     * @param payload obtained via encodeABI() in web3
-     * @return the data being returned by the ERC725 Account
+     * @dev Allows any address (executor) to execute a payload (= abi-encoded function call) in the linked {target} given they have a signed message from
+     * a controller with some permissions.
```
Comment (b00ste) [inline contracts/LSP6KeyManager/ILSP6KeyManager.sol:76]: Maybe we can remove the line break and also add `@notice`?

**Comment by CJ42** [comment]
@b00ste I have added `@notice` tags to all custom errors 🫡

**Review by b00ste — APPROVED** [review]

**Pull request merged**